### PR TITLE
Fixes for #31, #32, #33

### PR DESCRIPTION
**Pull request by crudo10**

### CODE REVIEW

**Comment by beanqueen** [comment]
i think you need to update your PR, since "There are no new changes to show."

File: swagger_parser/swagger_parser.py [hunk 1]
```diff
@@ -71,6 +71,11 @@ def __init__(self, swagger_path=None, swagger_dict=None, swagger_yaml=None, use_
         except Exception as e:
             raise ValueError('{0} is not a valid swagger2.0 file: {1}'.format(swagger_path,  e))
 
+            validate_spec(self.specification, '')
```
Comment (cyprieng) [inline swagger_parser/swagger_parser.py]: I don't get the utility of this

File: swagger_parser/swagger_parser.py [hunk 2]
```diff
@@ -194,7 +199,7 @@ def get_example_from_prop_spec(self, prop_spec):
             return self._get_example_from_basic_type(prop_spec['type'][0])[0]
 
         # Default - basic type
-        logging.info("falling back to basic type, no other match found")
+        # logging.info("falling back to basic type, no other match found")
```
Comment (cyprieng) [inline swagger_parser/swagger_parser.py]: Can you keep the logging please ?

File: swagger_parser/swagger_parser.py [hunk 3]
```diff
@@ -228,18 +233,19 @@ def _get_example_from_properties(self, spec):
...
-            if inner_name not in required:
-                continue
-            partial = self.get_example_from_prop_spec(inner_spec)
-            # While get_example_from_prop_spec is supposed to return a list,
-            # we don't actually want that when recursing to build from
-            # properties
-            if isinstance(partial, list):
-                partial = partial[0]
-            example[inner_name] = partial
+        if properties != None:
```
Comment (cyprieng) [inline swagger_parser/swagger_parser.py]: It is better to use `is not None` instead of `!= None`

**Comment by cyprieng** [comment]
The tests are failing due to lint error: https://travis-ci.org/Trax-air/swagger-parser/jobs/178064560

**Comment by cyprieng** [comment]
Fix #31 #32 #33

File: swagger_parser/swagger_parser.py (hunk 1 above)
Comment (crudo10) [inline swagger_parser/swagger_parser.py]: Me neither... must be left over from my debugging, sorry

File: swagger_parser/swagger_parser.py (hunk 3 above)
Comment (crudo10) [inline swagger_parser/swagger_parser.py]: Agreed

**Comment by cyprieng** [comment]
Line 872 and 875 are too long, max length is 120

**Comment by cyprieng** [comment]
Good the tests are green :smile: 
Can you add one test for each of your fix (eg for each of your issues) ?

**Comment by crudo10** [comment]
I am not sure I am familiar with the test strategy. Do I need to modify some data (swagger) sample to introduce test cases that caused bugs? Or I can do that at UT level?

**Comment by cyprieng** [comment]
I think the best option is to call the functions you just fix with args that was failing before your test, and check that is working now. You can put this either in an already existing test case (function in this file: https://github.com/Trax-air/swagger-parser/blob/master/tests/test_swagger_parser.py) if one is already testing the function, or create a new test case.

File: swagger_parser/swagger_parser.py [hunk 4]
```diff
@@ -791,7 +801,14 @@ def get_response_example(self, resp_spec):
...
             elif 'items' in resp_spec['schema'] and resp_spec['schema']['type'] == 'array':  # Array
-                definition_name = self.get_definition_name_from_ref(resp_spec['schema']['items']['$ref'])
+                if '$ref' in resp_spec['schema']['items']:
+                    definition_name = self.get_definition_name_from_ref(resp_spec['schema']['items']['$ref'])
+                else:
+                    if 'type' in resp_spec['schema']['items']:
+                        definition_name = self.get_definition_name_from_ref(resp_spec['schema']['items']['type'])
+                        return [definition_name]
+                    else:
+                        resp_spec['schema']['items']
```
Comment (beanqueen) [inline swagger_parser/swagger_parser.py]: what does this line do ?

**Comment by beanqueen** [comment]
we might want to consider some refactoring of the code to reduce the number of indentation levels and enhance readability

File: swagger_parser/swagger_parser.py [hunk 5]
```diff
@@ -791,7 +801,14 @@ def get_response_example(self, resp_spec):
                 definition_name = self.get_definition_name_from_ref(resp_spec['schema']['$ref'])
                 return self.definitions_example[definition_name]
             elif 'items' in resp_spec['schema'] and resp_spec['schema']['type'] == 'array':  # Array
-                definition_name = self.get_definition_name_from_ref(resp_spec['schema']['items']['$ref'])
+                if '$ref' in resp_spec['schema']['items']:
+                    definition_name = self.get_definition_name_from_ref(resp_spec['schema']['items']['$ref'])
+                else:
+                    if 'type' in resp_spec['schema']['items']:
```
Comment (beanqueen) [inline swagger_parser/swagger_parser.py]: this is the only time you check if 'type' is in resp_spec['schema']['items'], all other times you assume it's there

Comment (crudo10) [inline swagger_parser/swagger_parser.py]: Yep I have missed that

**Comment by cyprieng** [comment]
Sorry for the really long delay. I will check this soon :smiley:

**Comment by philippeluickx** [comment]
Running into some errors which would be solved with this PR. Would be awesome to get this to master!

File: swagger_parser/swagger_parser.py [hunk 6]
```diff
@@ -791,7 +801,15 @@ def get_response_example(self, resp_spec):
...
+                        definition_name = self.get_definition_name_from_ref(resp_spec['schema']['items'])
+                        return [definition_name]
+                    else:
+                        # logging.warn("No item type in: " + resp_spec['schema'])
```
Comment (cyprieng) [inline swagger_parser/swagger_parser.py]: Can you remove this line ?

**Comment by flavianh** [comment]
I resolved the conflict and kept the logging line for safety

**Comment by flavianh** [comment]
@philippeluickx I'm merging this.
@cyprieng I'll fix the tests in a next PR to drop python2.6 support. I'll make a major version bump then

**Pull request merged**